telethon/tl/custom/dialog.py

```diff
@@ -96,6 +100,13 @@ def __init__(self, client, dialog, entities, messages):
         return await self._client.send_message(
             self.input_entity, *args, **kwargs)
 
+    async def delete(self):
+        try:
+            await self._client(DeleteHistoryRequest(self, 0))
+
+        except PeerIdInvalidError:
```

Comment:
You have the `is_channel` property… why don't you use it instead making an invalid API call?